### PR TITLE
Blogroll: Disable blog appender sites that have been added to blog roll

### LINKED ISSUE #82122: Blogroll: Duplicate Recommendations for the Same Blog is Allowed
### Quick summary

Users can recommend a blog more than once, even if it's already added to their blogroll.

### Steps to reproduce

1. Add a Blogroll block. 
2. Add a Blogroll item and select a blog to recommend. 
3. Add another Blogroll item and select the same blog to recommend. 
4. Observe that you can recommend the same blog again.  

https://github.com/Automattic/wp-calypso/assets/67279475/1af8c6d0-edea-4677-832f-74c7a08b644e

### What you expected to happen

Once a blog is recommended and added to the Blogroll list, it should either be grayed out, removed from the list of recommendable blogs, or there should be a notification stating that the blog is already on the list.

### What actually happened

You can add the same blog multiple times.

### Impact

One

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

cc: @Automattic/vertex

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes https://github.com/Automattic/wp-calypso/issues/82122

## Proposed changes:

Disables sites in the blog appender that have already been added to the blogroll

CleanShot 2023-09-26 at 23 27 07@2x

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Checkout branch
* Go to editor and add the blogroll block
* Open the blog appender and select a suggested site. It should now be added to the blogroll.
* Open the blog appender again, verify the site you just added is disabled, and that the site cannot be added to the blogroll.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js [hunk 1]
```diff
@@ -12,7 +13,12 @@ export default function BlogrollAppenderResults( { results, showPlaceholder, onS
 			{ results.length > 0 && (
 				<ul aria-live="polite">
 					{ results.map( result => (
-						<li key={ result.blog_id }>
+						<li
+							key={ result.blog_id }
+							className={ classNames( 'jetpack-blogroll__appender-result-container', {
+								'is-disabled-result': result?.duplicateRecommendation,
+							} ) }
+						>
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js]: Let's change the `a` below into a button and disable it. Items that have `pointer-events: none` are still keyboard navigable and the site will be added when people tab-nav and press enter.

File: projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender/index.js [hunk 2]
```diff
@@ -14,11 +15,29 @@ export default function BlogrollAppender( { subscriptions, clientId } ) {
...
+		const itemBlogId = site?.blog_id;
+		const itemBlogrollItem = blogrollItems.find( item => {
+			return item?.attributes.id === itemBlogId;
+		} );
+		if ( itemBlogrollItem ) {
+			return { ...site, duplicateRecommendation: true };
+		}
+		return site;
+	} );
+
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender/index.js]: A cleaner way to get this info is like so:

```js
const [ siteRecommendations ] = useEntityProp(
	'root',
	'site',
	'Blogroll Recommendations'
);
```

We store this info on [every change](https://github.com/Automattic/jetpack/blob/8eef2eb53e93310e95eea1fd369048d85d72cf98/projects/plugins/jetpack/extensions/blocks/blogroll/use-site-recommendations.js#L10-L14).

And I would pass the items as-is and determine duplication inside the `BlogrollAppenderResults`. The Appender doesn't need to know about this logic.

**Review by alshakero — CHANGES_REQUESTED** [review]
Clean solution! I thought it would be more complex, thanks!

Left some comments.

File: projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
 
-			{ results.length === 0 && ! showPlaceholder && (
+			{ updatedResults.length === 0 && ! showPlaceholder && (
 				<div role="status">{ __( 'No websites found.', 'jetpack' ) }</div>
 			) }
-			{ results.length > 0 && (
+			{ updatedResults.length > 0 && (
 				<ul aria-live="polite">
-					{ results.map( result => (
+					{ updatedResults.map( result => (
```
Comment (alshakero) [inline projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js]: I would just do 

```suggestion
			const isDuplicate = siteRecommendations.find( sr => {
			return sr?.id === result?.blog_id;
		} )
					{ updatedResults.map( result => (
```

**Review by alshakero — DISMISSED** [review]
No need for `updatedResults`.

File: projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js (hunk 3 above)
Comment (agrullon95) [reply projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js]: I updated this. Much cleaner!

File: projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js [hunk 4]
```diff
@@ -18,26 +21,37 @@ export default function BlogrollAppenderResults( {
...
+							<li
+								key={ result.blog_id }
+								className={ classNames( 'jetpack-blogroll__appender-result-container', {
+									'is-disabled-result': isDuplicate,
+								} ) }
+							>
+								<button
+									className="jetpack-blogroll__appender-result-title"
+									disabled={ isDuplicate }
+									href="#"
```
Comment (heavyweight) [inline projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender-results/index.js]: We need to remove this

**Review by heavyweight — APPROVED** [review]

**Pull request merged**